### ISSUE TITLE
SnackBar not announced for a11y on iOS

### ISSUE THREAD

**Issue by jyygu** [flutter/flutter#41427]
Internal: b/141719626

When SnackBar is displayed, the message is not automatically announced on iOS.  We want this to be announced without user having to explicitly moving the focus, as it works on Android.

There is this documentation about [liveRegion](https://api.flutter.dev/flutter/semantics/SemanticsProperties/liveRegion.html), which talks about its usage in SnackBar.  But it's not clear why it is not working or intended to not work.

We currently don't have a workaround.

**Comment by jonahwilliams** [comment]
Originally I did not implement this on iOS because 1) iOS has no concept of a live region and 2) there was no official guidance that iOS snackbars should behave this way.

I took a look at a few possible options to supporting this automatically, through the framework and the engine and neither really works well due to the lack of accessibility APIs on iOS. My recommendation is to perform the announcement yourself using the existing flutter SemanticsService API:

```dart
void onShowSnackbar(String message) {
  Scaffold.of(context).showSnackBar(SnackBar(
     content: Text(message),
      duration: Duration(seconds: 2),
  ));
  if (defaultTargetPlatform == TargetPlatform.iOS) {
    SemanticsService.announce(message, textDirection: Directionality.of(context))
  }
}

```

**Comment by Hixie** [comment]
What does iOS MDC Snackbar do?

**Comment by Hixie** [comment]
I guess we should ask an iOS a11y expert what we should do.

**Comment by jonahwilliams** [comment]
The iOS MDC does the announcement, but its not clear why

**Comment by jkurtw** [comment]
Is this just waiting on the build to go green?
